circuits.go

```diff
@@ -1,42 +1,207 @@
+	AtomicQuerySigWithRelayCircuitID CircuitID = "credentialAtomicQuerySigWithRelay"
+)
+
+// ErrorCircuitIDNotFound returns if CircuitID is not registered
+var ErrorCircuitIDNotFound = errors.New("circuit id not supported")
+
+const (
+	defaultMTLevels       = 40 // max MT levels, default value for identity circuits
+	defaultValueArraySize = 16 // max value array size, default value for identity circuits
+)
+
+var circuitsRegistry = map[CircuitID]Data{}
 var circuitsLock = new(sync.RWMutex)
 
 // RegisterCircuit is factory for circuit init.
-// This is typically done during init() in the method's implementation
-func RegisterCircuit(id CircuitID, c BaseCircuit) {
+// This is done during init() in the method's implementation
+func RegisterCircuit(id CircuitID, c Data) {
 	circuitsLock.Lock()
 	defer circuitsLock.Unlock()
-	defaultCircuits[id] = c
+
+	circuitsRegistry[id] = c
+}
+
+// nolint // register supported circuit
+func init() {
+
+	RegisterCircuit(AuthCircuitID, Data{
+		Input:           AuthInputs{},
+		Output:          &AuthPubSignals{},
+		VerificationKey: embedFSLoader{"verificationKeys/auth.json"},
+		ProvingKey:      nil,
+	})
+
+	RegisterCircuit(StateTransitionCircuitID, Data{
+		Input:           StateTransitionInputs{},
+		Output:          &StateTransitionOutput{},
+		VerificationKey: embedFSLoader{"verificationKeys/stateTransition.json"},
+		ProvingKey:      nil,
+	})
+
+	RegisterCircuit(AtomicQueryMTPCircuitID, Data{
+		Input:           AtomicQueryMTPInputs{},
+		Output:          &AtomicQueryMTPPubSignals{},
+		VerificationKey: embedFSLoader{"verificationKeys/credentialAtomicQueryMTP.json"},
+		ProvingKey:      nil,
+	})
+
+	RegisterCircuit(AtomicQueryMTPWithRelayCircuitID, Data{
+		Input:           AtomicQueryMTPWithRelayInputs{},
+		Output:          &AtomicQueryMTPWithRelayPubSignals{},
+		VerificationKey: embedFSLoader{"verificationKeys/credentialAtomicQueryMTPWithRelay.json"},
+		ProvingKey:      nil,
+	})
+	RegisterCircuit(AtomicQuerySigCircuitID, Data{
+		Input:           AtomicQuerySigInputs{},
+		Output:          &AtomicQuerySigPubSignals{},
+		VerificationKey: embedFSLoader{"verificationKeys/credentialAtomicQuerySig.json"},
+		ProvingKey:      nil,
+	})
+}
+
+// BaseConfig base circuit's config, provides default configuration for default circuits
+type BaseConfig struct {
+	MTLevel        int // Max levels of MT
+	ValueArraySize int // Size if value array in identity circuits
+}
+
+// GetMTLevel max circuit MT levels
+func (c BaseConfig) GetMTLevel() int {
+	if c.MTLevel == 0 {
+		return defaultMTLevels
+	}
+	return c.MTLevel
+}
+
+// GetValueArrSize return size of circuits value array size
+func (c BaseConfig) GetValueArrSize() int {
+	if c.ValueArraySize == 0 {
+		return defaultValueArraySize
+	}
+	return c.ValueArraySize
+}
+
+// InputMarshaller interface implemented by types that can marshal circuit `input` structures
+type InputMarshaller interface {
+	CircuitInputMarshal() ([]byte, error)
```

Comment:
if we rename CircuitInputMarshal / CircuitOuputUnmarshall don't forget to change interfaces to Marshal/ Unmarshal